util/opts.rb

```diff
@@ -1,5 +1,28 @@
-module Opts
+class Opts
   FLAG_PREFIX = "--"
+  attr_reader :port, :char, :host, :default_color_id, :default_background_color_id, :template
+
+  def initialize(command_line_args)
+    config = parse_config(command_line_args)
+  
+    @port = config.fetch('port', 8000).to_i
+    @char = config['char']
+    @host = config.fetch('host', '127.0.0.1')
+    @default_color_id = config.fetch('default-color-id', 7).to_i
+    @default_background_color_id = config.fetch('default-background-color-id', 0).to_i
+    @template = config['template']
+  end
+
+  def parse_config(command_line_args)
```

Comment:
looks like there's a convention of `parse_flag` or `parse_command` that we may need to add here. If you can explain to me their purpose we can write some tests to cover this and add the implementation

Reply:
`parse_flag` is fairly obvious and that's what we're doing but `parse_command` i can revisit